proposals/0167-app-services.md

```diff
@@ -268,30 +268,109 @@ A weather service is defined as a service that can provide weather data.
+	        <description> This is the timestamp of when the data was generated. This is to ensure any time or distance given in the data can accurately be adjusted if necessary. </description>
+	    </param>
+	
+	    <param name="origin" type="LocationDetails" mandatory="false"/>
+	
+	    <param name="destination" type="LocationDetails" mandatory="false"/>
+	
+	    <param name="destinationETA" type="DateTime" mandatory="false"/>
+   
+       <param name="wayPoints" type="LocationDetails" mandatory="false" array="true">
+           <description> The current way points for the current navigation session. See also LocationDetails</description>
+        </param>
+   
+        <param name="instructions" type="NavInstruction" array="true" mandatory="false">
+            <description> This array should be ordered with the first object being the start location of the current trip and the last object the final destination. </description>
+	    </param>
+	
+	    <param name="nextInstructionETA" type="DateTime" mandatory="false"/>
+	    <param name="nextInstructionDistance" type="Float" mandatory="false">
+	        <description>The distance to this instruction from </description>
+	    </param>
+        <param name="nextInstructionDistanceScale" type="Float" mandatory="false">
+            <description>Distance till next maneuver (starting from) from previous maneuver.</description>
+        </param>
+	
+    </struct>
 
-	</struct>
+    <struct name="NavInstruction">
+	    <param name="locationDetails" type="LocationDetails" mandatory="true">
+	    </param>
+	
+	    <param name="action" type="NavAction" mandatory="true">
+	    </param>	
+	
+	    <param name="direction" type="int" minValue="0" maxValue="359" mandatory="false">
+	        <description>The angle at which this instruction takes place. For example, 0 would mean straight, 45 a sharp right, 180 a U-Turn, etc. </description>
+	    </param>
 	
+	    <param name="junctionType" type="NavJunctionType" mandatory="false">
+	    </param>	
+    </struct>
+    
+    <enum name="NavAction">
+        <element name="TURN">
+            <description> Using this action plush a supplied direction can give the type of turn. </description>
+        </element>
+        <element name="EXIT">
+        </element>
+        <element name="STAY">
+        </element>              
+        <element name="MERGE">
+        </element>
+        <element name="FERRY">
+        </element>        
+    <enum>
+    
+    <enum name="NavJunctionType">
+        <element name="REGULAR">
+            <description> A junction that represents a standard intersection with a single road crossing another.</description>
+        </element>
+        <element name="BIFURCATION">
+            <description> A junction where the road splits off into two paths; a fork in the road.</description>
+        </element>
+        <element name="MULTI_CARRIAGEWAY">
+            <description> A junction that has multiple intersections and paths</description>
```

Comment:
Update with period at the end to be consistent: 
A junction that has multiple intersections **and paths.**